__test__/server/api/campaign/campaign.test.js

```diff
@@ -415,14 +415,20 @@ describe("Caching", () => {
 
       queryLog = [];
       console.log("STARTING TEXTING"); // eslint-disable-line no-console
-      for (let i = 0; i < 5; i++) {
-        await sendMessage(testContacts[i].id, testTexterUser, {
-          userId: testTexterUser.id,
-          contactNumber: testContacts[i].cell,
-          text: "test text",
-          assignmentId
-        });
+
+      const waitFor = [];
```

Comment:
The linter complained about using `await` inside a loop so I changed it to use this pattern.

Reply:
That was a good call.  Can you tell me how you decided that it should be an empty array?

Reply:
I'm not sure I understand the question?

The idea about the linter complaint and the change was to take `await` out of the loop. We use an array to pass promises to `Promise.all()` after the loop. We need to start with an empty array to add the promises to.

The whole thing probably could have been done differently, maybe something like the following (as someone did elsewhere in the file)

```
Promise.all(testContacts.slice(0,5).map(contact => sendMessage(testContacts[i].id, testTexterUser, {
            userId: testTexterUser.id,
            contactNumber: testContacts[i].cell,
            text: "test text",
            assignmentId: assignmentId.toString()
          }))
```